src/Sulu/Bundle/AdminBundle/Resources/js/components/Table/Row.js

```diff
@@ -64,61 +99,111 @@ export default class Row extends React.PureComponent<Props> {
+            style.paddingLeft = (this.props.depth * this.props.depthPadding) + 'px';
+        }
+
+        if (this.props.selectInFirstCell) {
+            select = (
+                <div className={tableStyles.cellSelect}>
+                    {this.createSelect()}
+                </div>
+            );
+        }
+
+        return (
+            <div className={tableStyles.cellContent} style={style}>
+                {select}
+                {toggler}
+                {children}
+            </div>
+        );
+    };
+
+    createSelect = () => {
+        if (this.isSingleSelect()) {
+            return this.createRadioCell();
+        } else if (this.isMultipleSelect()) {
+            return this.createCheckboxCell();
+        }
+    };
+
+    createToggler = () => {
+        const {isLoading, expanded} = this.props;
 
         return (
-            <Cell
-                key={key}
-                small={true}
+            <span
+                onClick={expanded === false ? this.handleExpand : this.handleCollapse}
+                className={tableStyles.toggleIcon}
             >
-                <Radio
-                    skin="dark"
-                    value={identifier}
-                    checked={!!selected}
-                    onChange={this.handleSingleSelectionChange}
-                />
-            </Cell>
+                {isLoading &&
+                    <Loader size={10} />
+                }
+                {!isLoading &&
+                    <Icon name={expanded === true ? 'su-angle-down' : 'su-angle-right'} />
+                }
+            </span>
+        );
+    };
+
+    createRadioCell = () => {
+        const {selected} = this.props;
+
+        return (
+            <Radio
+                skin="dark"
+                value={this.getIdentifier()}
+                checked={!!selected}
+                onChange={this.handleSingleSelectionChange}
+            />
         );
     };
 
     createCheckboxCell = () => {
-        const {id, selected, rowIndex} = this.props;
-        const key = `checkbox-${rowIndex}`;
-        const identifier = id || rowIndex;
+        const {selected} = this.props;
 
         return (
-            <Cell
-                key={key}
-                small={true}
-            >
-                <Checkbox
-                    skin="dark"
-                    value={identifier}
-                    checked={!!selected}
-                    onChange={this.handleMultipleSelectionChange}
-                />
-            </Cell>
+            <Checkbox
+                skin="dark"
+                value={this.getIdentifier()}
+                checked={!!selected}
+                onChange={this.handleMultipleSelectionChange}
+            />
         );
     };
 
     createButtonCells = () => {
-        const {id, rowIndex} = this.props;
-        const {buttons} = this.props;
+        const {buttons, rowIndex} = this.props;
+        const identifier = this.getIdentifier();
```

Comment:
With the above mentioned improvement we could directly use `this.identifier` in the JSX.